api/v1alpha1/repository_types.go

```diff
@@ -28,14 +51,38 @@ type RepositorySpec struct {
 	// INSERT ADDITIONAL SPEC FIELDS - desired state of cluster
 	// Important: Run "make" to regenerate code after modifying this file
 
-	// Foo is an example field of Repository. Edit repository_types.go to remove/update
-	Foo string `json:"foo,omitempty"`
+	// URL chart repository address
+	// +kubebuilder:validation:Required
+	URL string `json:"url"`
+
+	// BasicAuthSecret if the chart repository requires basic auth authentication,
+	// set the username and password to secret, with the fields user and password respectively.
+	BasicAuthSecret string `json:"basicAuthSecret,omitempty"`
+
```

Comment:
maybe add PullStategy ?
```go
// pullStategy for this repository
PullStategy *PullStategy `json:"pullStategy,omitempty"`

// PullStategy for pulling components in repository
type PullStategy struct {
	// Interval for pulling
	IntervalSeconds int `json:"intervalSeconds,omitempty"`
	// Timeout for pulling
	TimeoutSeconds int `json:"timeoutSeconds,omitempty"`
	// Retry upon timeout
	Retry int `json:"retry,omitempty"`
}

```

Reply:
@0xff-dev any thought here?  I think this depends on how you implement this `repo watch`